### PR TITLE
Security Utility Knife

### PR DESCRIPTION
**Pull request by Arahimine**
[OBJECTS] [FEATURE] [SPRITES]
## About the PR
This PR adds a utility knife to the Security weapons vendor, available for 1 utility token. Comes with a custom suicide.

First and foremost, this is not designed for combat, and should only be used as such as a last resort. At 7 physical damage and 10 stamina damage, it's firmly weaker than a harmbaton, the only conceivable advantage being the trivial (equal to wirecutters) bleeding. Instead, true to its name, this knife is for UTILITY, and has 3 functions to that end:

- It functions as a crowbar, being able to pry firelocks, depowered doors, and floor tiles (but not fling them)
- It deals bonus damage to kudzu, being much better than a harmbaton for that purpose
- It can be stored inside your military boots - this includes variants such as the HoS heavy military boots. Boots with an item stowed in them will have their examine text slightly adjusted to indicate

They also, obviously, function as a knife for all cutting applications. Is it possible to cut a pizza and look tacticool while doing it? Find out!

Featuring incredible spritework by ckey Durin the Deathless:
image

Three variants (generic, NT, and Syndicate) are included in the PR, but aren't available in-game without admin shenanigans. In the future, I'd like to see about adding these somewhere (giving the NT one to NTSCs, perhaps)
image 
image
image

Additionally, to ~~justify~~ show off the item stowing code, clown shoes can now also stow bike horns.
## Technical info
To achieve these effects, a few things have been done.
- Items now have a 'kudzu_force' variable that overrides their usual 'force' in the context of kudzu damage calculations as long as it's not null (null by default)
- Crowbars and prying tools (incl. the utility knife) can no longer pry floor tiles that have kudzu on top of them - this made it really hard to fight kudzu using them and also didn't make much sense anyway
- The item stowing code (used to store the knife inside boots) is defined at the /obj/item/clothing/shoes level, and this functionality can easily be extended to any other shoes in the game by just defining the variable 'stowable_items' (being a list containing all the valid stowable item types for those shoes)

This is my first time coding in BYOND in well over a year, so forgive me if I made any faux pases in these things and feel free to correct any that I did. I'm not sure how normal it is to define a variable as null in any context, but it doesn't throw any warnings and I've extensively bug-tested it.

## Why's this needed?
Tacticool! Mostly, a crowbar is a pretty understandable item for Security (and the crew in general) to carry on them for emergencies (nobody likes being completely defeated by a single firelock, after all), so this is a way for SecOffs to do just that in style.

Balance was a serious consideration - as mentioned in the introduction, a harmbaton is still better in any combat scenario, as the trivial bleed damage won't let you win any fights that you weren't going to win anyway. The kudzu bonus is not *too* impactful, as anyone with experience knows that the difficulty in fighting kudzu isn't what weapons you have available, but how many people you have to wield them. (also it seems a little odd that Security lacks a solid answer to kudzu beyond just taking tools from the other departments)

## Changelog

```changelog
(u)Arahimine
(*)Added a utility knife to the Security weapons vendors for 1 utility token, best used for prying open firelocks and cutting down kudzu. Try slotting it inside your military boots! Sprite by Durin the Deathless.
(+)Bike horns can now be stowed inside clown shoes.
```

### CODE REVIEW

**Comment by CaioFrancisco** [comment]
> Additionally, to justify show off the item stowing code, clown shoes can now also stow bike horns.

extremely unbalanced. pls nerf.

**Comment by pali6** [comment]
Our style direction for security is generally to stay away from tacticool, lethal or lethal-looking things. I also feel like this knife would get misunderstood to be a good weapon in its current form.

**Comment by Azrun** [comment]
> It deals bonus damage to kudzu, being much better than a harmbaton for that purpose
> The kudzu bonus is not too impactful, as anyone with experience knows that the difficulty in fighting kudzu isn't what weapons you have available, but how many people you have to wield them. (also it seems a little odd that Security lacks a solid answer to kudzu beyond just taking tools from the other departments)

I think Botany should be the one with tools to deal with kudzu. And Botany should be able to take part in dispensing said tools. Spotlight the relevant department.

**Comment by Glamurio** [comment]
> Our style direction for security is generally to stay away from tacticool, lethal or lethal-looking things. I also feel like this knife would get misunderstood to be a good weapon in its current form.

I really enjoy the item idea, and Security often having to navigate maintenance or unpowered rooms to chase bad guys could definitely do with an "on-spawn" crowbar. Perhaps we could rework the look to look more like a Swiss Army Knife? More prominently showcasing the crowbar or snipping bits, less so the knife bit. Could also make it a wirecutter fittingly, but Security doesn't really need those.

Basically, item cool, maybe rework the sprite or make it less weapon-looking. Since it's mechanically just an "anti-kudzu crowbar", perhaps that could inspire a new design.

**Comment by Arahimine** [comment]
(Sorry for the late reply!)

> Our style direction for security is generally to stay away from tacticool, lethal or lethal-looking things. I also feel like this knife would get misunderstood to be a good weapon in its current form.

I completely understand that! Ensuring that the non-lethality got across was and still is one of my main design concerns. With that in mind, how about this for a revised sprite?
image
The rectangular edge is inspired by real-world utility knives, and should get across that this is a tool, not a weapon. It's also worth noting that the description of the item in the vendor explicitly mentions 'non-combat'.

> I think Botany should be the one with tools to deal with kudzu. And Botany should be able to take part in dispensing said tools. Spotlight the relevant department.

I agree in a broad sense about the role of the crew, but in this case, I don't think that Botanist is proactive enough of a role to justify limiting the ability of other departments to respond to threats. (Also, they still absolutely can do that - this knife doesn't beat atrazine in terms of effectiveness, and Botany has the easiest access to that of any department.)
